l3kernel/l3str.dtx

```diff
@@ -787,6 +787,13 @@
 %   not to \texttt{\i}).
 % \end{function}
 %
+% \begin{function}[added = 2023-05-19, EXP]{\str_mdfive_hash:n}
+%   \begin{syntax}
+%     \cs{str_mdfive_hash:n} \Arg{str}
+%   \end{syntax}
+%   Expands to the MD5 sum generated from the \meta{str}.
```

Comment:
Similar to `\str_if_eq:nnTF` and `\str_if_eq:eeTF` ([doc](https://github.com/latex3/latex3/blob/139c10f8930315b7c27f8000fea839d7e9ef72fb/l3kernel/l3str.dtx#L247-L266), [code](https://github.com/latex3/latex3/blob/139c10f8930315b7c27f8000fea839d7e9ef72fb/l3kernel/l3str.dtx#L1169-L1180)), I think the argument of `\str_mdfive_hash:n` need not already be string, but can be a general token list.
```tex
\cs{str_if_eq:nnTF} \Arg{tl_1} \Arg{tl_2} \Arg{true code} \Arg{false code}
```

But then
```tex
\cs_new:Npn \str_mdfive_hash:n #1
  { \exp_not:Nw \tex_mdfivesum:D \exp_not:Nw { \tl_to_str:n {#1 } } }
\cs_new:Npn \str_mdfive_hash:e #1 { \tex_mdfivesum:D {#1} }
```

Reply:
I was aware of the `\Arg{tl_1}` part in `\str_if_eq:nnTF` doc when opening #1210 (hence `\cs{str_mdfive_hash:n} \Arg{token list}` was used in the attached example there), but didn't get a full understanding about how to understand the differences between  `\str_` module name and `tl_1`/`tl_2` arg spec until just then, after reading its implementation.
- `\str_xxx:n` accepts a token list but converts it to string before calling the primitive,
- `\str_xxx:e` calls the primitive directly and leaves the tl2str conversion to primitive itself.